src/main/java/org/spongepowered/api/item/inventory/Inventory.java

```diff
@@ -498,6 +534,15 @@ default Inventory transform(InventoryTransformation transformation) {
         return transformation.transform(this);
     }
 
+    /**
+     * Returns this inventory as an interactable inventory if possible.
+     *
+     * <p>Not all inventories are interactable (e.g. a custom inventory of size 5x5)</p>
+     *
+     * @return This inventory as an interactable inventory if possible.
+     */
+    Optional<Interactable> asInteractable();
+
```

Comment:
Maybe in case of a large grid inventory you can also provide a method like `asInteractableView` which takes the min and max coordinates to create view for.
E.g. for a 50x50 inventory, you can get a 3x9 view starting at row 4 and column 9, and display that to the player
Just an idea though

Reply:
I'd rather not add features with this. I'll have enough work to fix impl.
But I'll keep this in mind for later